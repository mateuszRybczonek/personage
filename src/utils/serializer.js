/* eslint-disable import/prefer-default-export */
export const serializeCards = cardsObj => Object.keys(cardsObj).reduce((acc, category) => {
  const cardsFromCategory = cardsObj[category];
  const cardsArr = Object.keys(cardsFromCategory).map(keyword => ({
    keyword,
    category,
    forbiddenWords: cardsFromCategory[keyword],
  }));

  return [
    ...acc,
    ...cardsArr,
  ];
}, []);
