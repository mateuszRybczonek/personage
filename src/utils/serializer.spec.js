import { serializeCards } from './serializer';

describe('serializeCards', () => {
  it('serializes object to proper array', () => {
    const cards = {
      category1: {
        lorem: ['a', 'b', 'c'],
        ipsum: ['d', 'e', 'f'],
      },
      category2: {
        dolor: ['g', 'h'],
        sit: ['i', 'j'],
      },
    };

    expect(serializeCards(cards)).toMatchSnapshot();
  });
});
