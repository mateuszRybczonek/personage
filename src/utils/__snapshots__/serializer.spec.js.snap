// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializeCards serializes object to proper array 1`] = `
Array [
  Object {
    "category": "category1",
    "forbiddenWords": Array [
      "a",
      "b",
      "c",
    ],
    "keyword": "lorem",
  },
  Object {
    "category": "category1",
    "forbiddenWords": Array [
      "d",
      "e",
      "f",
    ],
    "keyword": "ipsum",
  },
  Object {
    "category": "category2",
    "forbiddenWords": Array [
      "g",
      "h",
    ],
    "keyword": "dolor",
  },
  Object {
    "category": "category2",
    "forbiddenWords": Array [
      "i",
      "j",
    ],
    "keyword": "sit",
  },
]
`;
